blog/2023-07-3.4-3.13.md

```diff
@@ -0,0 +1,191 @@
+- While we were at it, we’ve also fixed [resuming in single-chunk uploads](https://github.com/transloadit/uppy/pull/4528), andimproved [chunk size calculation](https://github.com/transloadit/uppy/pull/4508).
+- [Golden Retriever integration with S3 Multipart](https://github.com/transloadit/uppy/pull/4526) was fixed.
+
+Additionally, we’ve disabled pause-resume buttons in the UI for remote S3 Multipart uploads — they are not supported on Companion, so the UI now reflects that.
+
+## Core
+
+Performance of adding and uploading big batches of files has (yet again) been greatly improved with refactors in both Uppy Core and Provider Views (that’s what we internally call all remote UIs: Instagram, Unsplash, Dropbox). 
+
+We’ve been able to achieve this by [refactoring out totals validation for Restrictions (maxTotalFileSize, maxNumberOfFiles)](https://github.com/transloadit/uppy/pull/4402): instead of doing it for each file, we perform the check at the end, after all the files have been added/validated. This was confirmed to work with 10k+ files! 🚀
+
+<img src="/img/blog/3.4-3.13/10k-files.jpg" className="border" alt="Uppy UI with over 10k files selected from a remote source" />
+
+We’ve also fixed an issue with delayed throttled progress events, which could lead to incorrect progress.
+
+## Transloadit
+
+### assemblyOptions
+
+We’ve introduced a new option called [`assemblyOptions: object | function`](https://uppy.io/docs/transloadit/#assemblyoptions), which replaces the `getAssemblyOptions`, `params` and `fields` (those are now deprecated and will be removed in the next major).
+
+Here’s how you can use the new option as an object:
+
+```js
+uppy.use(Transloadit, {
+  assemblyOptions: {
+    params: {
+      auth: { key: 'key-from-transloadit' },
+      template_id: 'id-from-transloadit',
+      steps: {
+        // Overruling Template at runtime
+      },
+    },
+    signature: 'generated-signature',
+    fields: {
+      // Dynamic or static fields to send along
+    },
+  }
+})
+```
+
+And here’s an example with a function, to be able to set meta fields per-file, for instance:
+
+```js
+uppy.use(Transloadit, {
+  assemblyOptions (file) {
+    return {
+      params: {
+        auth: { key: 'TRANSLOADIT_AUTH_KEY_HERE' },
+        template_id: 'xyz',
+      },
+      fields: {
+        caption: file.meta.caption,
+      },
+    }
+  }
+})
+```
+
+### Server-sent events
+
+Historically Transloadit supported progress updates via Socket.io. It is a robust and stable package, but came with a “price-tag” in the form of bundle size: [249.3 kB Minified](https://bundlephobia.com/package/socket.io@4.7.1). So it’s been on our minds for a while to replace it with something more lightweight, without breaking backwards-compatibility for the older clients (so simply removing Socket.io on the server in favor of plain WebSockets is not desireable).
```

Comment:
You're checking out the server.

```suggestion
Historically Transloadit supported progress updates via Socket.io. It is a robust and stable package, but came with a “price-tag” in the form of bundle size: [38.3 kB Minified](https://bundlephobia.com/package/socket.io-client@4.7.1). So it’s been on our minds for a while to replace it with something more lightweight, without breaking backwards-compatibility for the older clients (so simply removing Socket.io on the server in favor of plain WebSockets is not desireable).
```

Reply:
Ohh haha, I thought it was a bit much :)) Sorry, thanks for catching!